pandas/tests/io/json/test_readlines.py

```diff
@@ -20,42 +21,63 @@ def lines_json_df():
     return df.to_json(lines=True, orient="records")
 
 
-def test_read_jsonl():
+def test_read_jsonl(request, engine):
     # GH9180
-    result = read_json('{"a": 1, "b": 2}\n{"b":2, "a" :1}\n', lines=True)
+    if engine == "pyarrow":
+        reason = "Pyarrow only supports a file path as an input and line delimited json"
+        request.node.add_marker(pytest.mark.xfail(reason=reason))
+
+    result = read_json('{"a": 1, "b": 2}\n{"b":2, "a" :1}\n', lines=True, engine=engine)
     expected = DataFrame([[1, 2], [1, 2]], columns=["a", "b"])
     tm.assert_frame_equal(result, expected)
 
 
-def test_read_datetime():
+def test_read_jsonl_engine_pyarrow(json_dir_path, engine):
+    result = read_json(
+        os.path.join(json_dir_path, "line_delimited.json"),
```

Comment:
```suggestion
        datapath("io", "json", "data", "line_delimited.json")
```

And then you can replace `json_dir_path` with `datapath` and remove `json_dir_path` from `pandas/tests/io/json/conftest.py`